javalib/src/main/scala/java/util/logging/Logger.scala

```diff
@@ -0,0 +1,252 @@
+package java.util.logging
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+object Logger {
+  val GLOBAL_LOGGER_NAME: String =  "global"
+
+  // Not implemented, deprecated on JDK 1.8
+  //val global: Logger
+
+  private val defaultLogLevel = Level.ALL
+
+  private val loggers: mutable.Map[String, Logger] = mutable.Map.empty
+
+  // Root is not visible to the outside but gives defaults
+  private[this] val rootLogger =
+    new Logger("", None, Some(defaultLogLevel), false, None)
+
+  private[this] val globalLogger =
+    new Logger(GLOBAL_LOGGER_NAME, None, None, true, Some(rootLogger))
+
+  def getGlobal():Logger = globalLogger
+
+  def getLogger(name: String): Logger = {
+    if (name == null)
+      throw new NullPointerException("Logger name cannot be null")
+    loggers.getOrElseUpdate(name,
+      new Logger(name, None, None, true, Some(rootLogger)))
+  }
+
+  // Not implemented, no resource bundle in scala.js
+  //def getLogger(name: String, resourceBundle: String): Logger
+
+  protected def findParent(logger: Logger): Option[Logger] = {
+    def go(s: List[String]): Option[Logger] = s match {
+      case Nil                                    =>
+        None
+      case b if loggers.contains(b.mkString(".")) =>
+        loggers.get(b.mkString("."))
+      case b                                      =>
+        go(b.dropRight(1))
+    }
+    go(Option(logger.getName).getOrElse("").split("\\.").toList.dropRight(1))
+  }
+
+  def getAnonymousLogger():Logger =
+    // No references to anonymous loggers are kept
+    new Logger(null, None, None, true, Some(rootLogger))
+
+  // Not implemented, no resource bundle in scala.js
+  //def getAnonymousLogger(resourceBundle: String):Logger
+}
+
+class Logger private (protected val name: String,
```

Comment:
`name, `resourceBundle` and the rest of this constructors `var`s should not be protected fields of `Logger`.